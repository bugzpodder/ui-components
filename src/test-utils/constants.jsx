// @flow
import React from "react";
import { EDC, PIPELINE } from "@grail/lib";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const GRAIL = "https://grail.com/wp-content/uploads/2016/12/cropped-final-favicon-32x32.png";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 1000; i++) {
  // Create about 17,000 records.
  LARGE_EXAMPLE_TABLE_DATA.push(...EXAMPLE_TABLE_DATA);
}

export const EXTRA_LARGE_EXAMPLE_TABLE_DATA = [];
for (let i = 0; i < 100; i++) {
  // Create about 1,700,000 records
  EXTRA_LARGE_EXAMPLE_TABLE_DATA.push(...LARGE_EXAMPLE_TABLE_DATA);
}

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];

export const ALPHABET_ANIMALS_DATA: Array<Array<?string>> = [
  ["alligator", "bat", "cat", "dog", "elephant", "fox"],
  ["goat", "horse", "iguana", "jaguar", "kangaroo", "lion"],
  ["monkey", "narwhal", "octopus", "penguin", "quail", "raccoon"],
  ["seal", "tiger", "umbrella bird", "vulture", "whale", "x-ray tetra"],
  ["yak", "zebra", null, null, null, null],
];

export const TEST_EXTERNAL_DOMAINS: Map<string, string> = new Map().set(EDC, "edc.com").set(PIPELINE, "pipeline.com");
