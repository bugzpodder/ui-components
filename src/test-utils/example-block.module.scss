.container {
  margin: 1rem 0;

  .header {
    font-family: monospace;
  }

  .text {
    background: #f6f8fa;
    font-size: 1rem;
    padding: .5rem;
  }
}