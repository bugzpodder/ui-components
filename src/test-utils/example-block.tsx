import React from "react";
import styles from "./example-block.module.scss";

type Props = {
  content: any;
  strongHeader: string;
  helperText?: string;
};

export const ExampleBlock = (props: Props) => {
  const { strongHeader, helperText = "", content } = props;
  return (
    <div className={styles.container}>
      <pre data-testid="example" className={styles.text}>
        <strong>{strongHeader}</strong>
        {helperText}= {JSON.stringify(content, null, 2)}
      </pre>
    </div>
  );
};
