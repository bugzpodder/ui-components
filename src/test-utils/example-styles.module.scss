.container {
  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.spinnerContainer {
  position: relative;

  .buttonContainer {
    display: inline-block;
      position: relative;
  }

  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.avatarContainer {
  .inLine {
    display: inline-block;
  }
}

.selectContainer {
  .commonSelect {
    max-width: 350px;
  }
}

.pagedTable {
  height: 600px;
}
