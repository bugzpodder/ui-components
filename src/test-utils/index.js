// @flow
export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
