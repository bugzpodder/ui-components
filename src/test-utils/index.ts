export {
  ALPHABET_ANIMALS_DATA,
  ADAMS_DATA,
  COUNTRIES,
  COUNTRY_NAMES,
  EMU_DATA,
  EXTRA_LARGE_EXAMPLE_TABLE_DATA,
  LARGE_EXAMPLE_TABLE_DATA,
  EXAMPLE_TABLE_DATA,
  GRAIL,
  TEST_EXTERNAL_DOMAINS,
} from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
export { StyleWrapper as ExampleWrapper } from "../style-wrapper";
