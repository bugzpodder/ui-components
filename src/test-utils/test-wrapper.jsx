// @flow
import React, { type Node } from "react";
// $FlowFixMe: there is no createGenerateClassName export in @material-ui/core/styles
import { StyleWrapper } from "../style-wrapper";
import { StylesProvider, createGenerateClassName } from "@material-ui/styles";

type Props = {
  children: Node<*>,
};

// TODO(nsawas): Figure out why this new method of disabling global styles
// isnt exactly working. For now we disable all styles with `disableGeneration` prop below.
const generateClassName = createGenerateClassName({
  disableGlobal: true,
});

class MockResizeObserver {
  observe = () => {};
  disconnect = () => {};
}

window.ResizeObserver = MockResizeObserver;

export const TestWrapper = (props: Props) => {
  const { children } = props;
  return (
    <StylesProvider
      disableGeneration
      generateClassName={generateClassName}
    >
      <StyleWrapper>{children}</StyleWrapper>
    </StylesProvider>
  );
};
