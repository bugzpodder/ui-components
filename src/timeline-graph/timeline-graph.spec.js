// @flow
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import mockConsole from "jest-mock-console";
import moment from "moment-timezone";
import { TabbedTimelineCard, TimelineCard, TimelineGraph } from ".";
import { TestWrapper } from "../test-utils";
import { cleanup, fireEvent, render } from "@testing-library/react";

afterEach(cleanup);
moment.tz.setDefault("America/Los_Angeles");
const rows = [
  {
    date: "2018-04-20T16:20:00",
    content: <div>Testing</div>,
  },
  {
    date: "2018-04-20T16:20:00",
    content: <div>Testing</div>,
    user: "nsawas",
  },
];

const tabContents = {
  tabOne: { content: rows },
  tabTwo: {
    content: [
      {
        date: "2019-04-20T16:20:00",
        content: <div>Test Two</div>,
      },
    ],
  },
};

test("render simple timeline graph", () => {
  const { container, getByTestId } = render(
    <TestWrapper>
      <TimelineGraph rows={rows} />
    </TestWrapper>,
  );
  expect(getByTestId("timeline-year-0")).toHaveTextContent("2018");
  expect(getByTestId("timeline-date-0")).toHaveTextContent("20 Apr");
  expect(container).toMatchSnapshot();
});

test("render simple timeline graph with time", () => {
  const { container, getByTestId } = render(
    <TestWrapper>
      <TimelineGraph
        rows={rows}
        isTimeVisible
      />
    </TestWrapper>,
  );
  expect(getByTestId("timeline-time-0")).toHaveTextContent("16:20");
  expect(container).toMatchSnapshot();
});

test("test tabbed timeline cartd", () => {
  const { container } = render(
    <TestWrapper>
      <TabbedTimelineCard tabContents={tabContents} />
    </TestWrapper>,
  );
  expect(container).toMatchSnapshot();
});

test("test invalid timeline card", () => {
  mockConsole();
  expect(() =>
    render(
      <TestWrapper>
        <TabbedTimelineCard />
      </TestWrapper>,
    ),
  ).toThrowError();
  expect(console.error).toHaveBeenCalled();
});

test("test selecting an item on the timeline card", () => {
  const mockOnSelect = jest.fn(result => result);
  const { container, getByTestId } = render(
    <TestWrapper>
      <TimelineCard
        onSelect={mockOnSelect}
        rows={rows}
      />
    </TestWrapper>,
  );
  fireEvent.click(getByTestId("timeline-item-0"));
  expect(mockOnSelect.mock.results[0].value).toEqual(0);
  expect(container).toMatchSnapshot();
});

test("test unselecting an item on the timeline card", () => {
  const mockOnSelect = jest.fn(result => result);
  const { container, getByTestId } = render(
    <TestWrapper>
      <TimelineCard
        isDayVisible
        classes={{
          root: "naji",
        }}
        onSelect={mockOnSelect}
        selectedItem={0}
        rows={rows}
      />
    </TestWrapper>,
  );

  // TODO(nsawas): write tests for all classes.
  expect(getByTestId("timeline-day-0")).toBeInTheDocument();
  expect(getByTestId("timeline-day-0")).toHaveTextContent("Fri");
  fireEvent.click(getByTestId("timeline-item-0"));
  expect(mockOnSelect.mock.results[0].value).toEqual(null);
  expect(container).toMatchSnapshot();
});
