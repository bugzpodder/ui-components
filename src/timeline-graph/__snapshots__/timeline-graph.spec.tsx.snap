// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render simple timeline graph 1`] = `
<div>
  <div
    class="timelineContainer"
  >
    <div
      class="timelineContent"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="timelineDateContainer"
        >
          <p
            class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-year-0"
          >
            2018
          </p>
          <div
            class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-date-0"
          >
            20 Apr
          </div>
        </div>
        <div
          class="timeline timelineStart"
        >
          <div
            class="notch"
          />
          <div
            class="MuiPaper-root timelinePaper MuiPaper-elevation6 MuiPaper-rounded"
            data-testid="timeline-item-0"
            id="0"
          >
            <div
              class="paperContent"
            >
              <div>
                Testing
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root lastItem MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="timelineDateContainer"
        >
          <p
            class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-year-1"
          >
            2018
          </p>
          <div
            class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-date-1"
          >
            20 Apr
          </div>
        </div>
        <div
          class="timeline timelineEnd"
        >
          <div
            class="notch"
          />
          <div
            class="MuiPaper-root timelinePaper MuiPaper-elevation6 MuiPaper-rounded"
            data-testid="timeline-item-1"
            id="1"
          >
            <div
              class="paperContent"
            >
              <div>
                Testing
              </div>
              <div
                class="user"
              >
                <div
                  class="avatarIconContainer"
                  data-testid="avatar-icon"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root userButton MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    data-testid="timeline-avatar"
                    disabled=""
                    id="timeline-avatar"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root userAvatar"
                        data-testid="avatar-icon-avatar"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                nsawas
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`render simple timeline graph with time 1`] = `
<div>
  <div
    class="timelineContainer"
  >
    <div
      class="timelineContent"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="timelineDateContainer"
        >
          <p
            class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-year-0"
          >
            2018
          </p>
          <div
            class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-date-0"
          >
            20 Apr
          </div>
          <p
            class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-time-0"
          >
            16:20
          </p>
        </div>
        <div
          class="timeline timelineStart"
        >
          <div
            class="notch"
          />
          <div
            class="MuiPaper-root timelinePaper MuiPaper-elevation6 MuiPaper-rounded"
            data-testid="timeline-item-0"
            id="0"
          >
            <div
              class="paperContent"
            >
              <div>
                Testing
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root lastItem MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="timelineDateContainer"
        >
          <p
            class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-year-1"
          >
            2018
          </p>
          <div
            class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-date-1"
          >
            20 Apr
          </div>
          <p
            class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
            data-testid="timeline-time-1"
          >
            16:20
          </p>
        </div>
        <div
          class="timeline timelineEnd"
        >
          <div
            class="notch"
          />
          <div
            class="MuiPaper-root timelinePaper MuiPaper-elevation6 MuiPaper-rounded"
            data-testid="timeline-item-1"
            id="1"
          >
            <div
              class="paperContent"
            >
              <div>
                Testing
              </div>
              <div
                class="user"
              >
                <div
                  class="avatarIconContainer"
                  data-testid="avatar-icon"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root userButton MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                    data-testid="timeline-avatar"
                    disabled=""
                    id="timeline-avatar"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root userAvatar"
                        data-testid="avatar-icon-avatar"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
                nsawas
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test selecting an item on the timeline card 1`] = `
<div>
  <div
    class="timelineContainer"
  >
    <div
      class="MuiPaper-root MuiCard-root timelineCard card MuiPaper-elevation6 MuiPaper-rounded"
      data-testid="card"
    >
      <div
        class="MuiCardContent-root"
        data-testid="card-body"
      >
        <div
          class="spinnerOverlay"
          data-is-active="false"
          data-testid="spinner-overlay"
        >
          <div
            class="loading"
          />
        </div>
        <div
          class="timelineContent"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="timelineDateContainer"
            >
              <p
                class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-year-0"
              >
                2018
              </p>
              <div
                class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-date-0"
              >
                20 Apr
              </div>
            </div>
            <div
              class="timeline timelineStart"
            >
              <div
                class="notch"
              />
              <div
                class="MuiPaper-root timelinePaperButton MuiPaper-elevation6 MuiPaper-rounded"
                data-testid="timeline-item-0"
                id="0"
              >
                <div
                  class="paperContent"
                >
                  <div>
                    Testing
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root lastItem MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="timelineDateContainer"
            >
              <p
                class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-year-1"
              >
                2018
              </p>
              <div
                class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-date-1"
              >
                20 Apr
              </div>
            </div>
            <div
              class="timeline timelineEnd"
            >
              <div
                class="notch"
              />
              <div
                class="MuiPaper-root timelinePaperButton MuiPaper-elevation6 MuiPaper-rounded"
                data-testid="timeline-item-1"
                id="1"
              >
                <div
                  class="paperContent"
                >
                  <div>
                    Testing
                  </div>
                  <div
                    class="user"
                  >
                    <div
                      class="avatarIconContainer"
                      data-testid="avatar-icon"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root userButton MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                        data-testid="timeline-avatar"
                        disabled=""
                        id="timeline-avatar"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root userAvatar"
                            data-testid="avatar-icon-avatar"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    nsawas
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test tabbed timeline cartd 1`] = `
<div>
  <div
    class="timelineContainer"
  >
    <div
      class="MuiPaper-root MuiCard-root timelineCard card MuiPaper-elevation6 MuiPaper-rounded"
      data-testid="card"
    >
      <div
        class="MuiCardContent-root timelineCardBody"
        data-testid="card-body"
      >
        <div
          class="MuiPaper-root MuiCard-root tabsWrapper MuiPaper-elevation1"
        >
          <div
            class="MuiTabs-root tabs"
            data-testid="card-tabs"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                class="MuiTabs-flexContainer tabsContainer"
                role="tablist"
              >
                <button
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
                  data-testid="timeline-tab-tabOne"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab One
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
                  data-testid="timeline-tab-tabTwo"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Tab Two
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-419 PrivateTabIndicator-colorPrimary-420 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </div>
        <div
          class="timelineContent"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="timelineDateContainer"
            >
              <p
                class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-year-0"
              >
                2018
              </p>
              <div
                class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-date-0"
              >
                20 Apr
              </div>
            </div>
            <div
              class="timeline timelineStart"
            >
              <div
                class="notch"
              />
              <div
                class="MuiPaper-root timelinePaper MuiPaper-elevation6 MuiPaper-rounded"
                data-testid="timeline-item-0"
                id="0"
              >
                <div
                  class="paperContent"
                >
                  <div>
                    Testing
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root lastItem MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="timelineDateContainer"
            >
              <p
                class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-year-1"
              >
                2018
              </p>
              <div
                class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-date-1"
              >
                20 Apr
              </div>
            </div>
            <div
              class="timeline timelineEnd"
            >
              <div
                class="notch"
              />
              <div
                class="MuiPaper-root timelinePaper MuiPaper-elevation6 MuiPaper-rounded"
                data-testid="timeline-item-1"
                id="1"
              >
                <div
                  class="paperContent"
                >
                  <div>
                    Testing
                  </div>
                  <div
                    class="user"
                  >
                    <div
                      class="avatarIconContainer"
                      data-testid="avatar-icon"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root userButton MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                        data-testid="timeline-avatar"
                        disabled=""
                        id="timeline-avatar"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root userAvatar"
                            data-testid="avatar-icon-avatar"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    nsawas
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test unselecting an item on the timeline card 1`] = `
<div>
  <div
    class="timelineContainer naji"
  >
    <div
      class="MuiPaper-root MuiCard-root timelineCard card MuiPaper-elevation6 MuiPaper-rounded"
      data-testid="card"
    >
      <div
        class="MuiCardContent-root"
        data-testid="card-body"
      >
        <div
          class="spinnerOverlay"
          data-is-active="false"
          data-testid="spinner-overlay"
        >
          <div
            class="loading"
          />
        </div>
        <div
          class="timelineContent"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="timelineDateContainer"
            >
              <p
                class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-year-0"
              >
                2018
              </p>
              <div
                class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-date-0"
              >
                20 Apr
                <div
                  data-testid="timeline-day-0"
                >
                  Fri
                </div>
              </div>
            </div>
            <div
              class="timeline timelineStart"
            >
              <div
                class="notch selected"
              />
              <div
                class="MuiPaper-root timelinePaperButton timelinePaperSelected MuiPaper-elevation4 MuiPaper-rounded"
                data-testid="timeline-item-0"
                id="0"
              >
                <div
                  class="paperContent"
                >
                  <div>
                    Testing
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root lastItem MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
          >
            <div
              class="timelineDateContainer"
            >
              <p
                class="MuiTypography-root smallDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-year-1"
              >
                2018
              </p>
              <div
                class="MuiTypography-root timelineDate MuiTypography-body1 MuiTypography-alignCenter"
                data-testid="timeline-date-1"
              >
                20 Apr
                <div
                  data-testid="timeline-day-1"
                >
                  Fri
                </div>
              </div>
            </div>
            <div
              class="timeline timelineEnd"
            >
              <div
                class="notch"
              />
              <div
                class="MuiPaper-root timelinePaperButton MuiPaper-elevation6 MuiPaper-rounded"
                data-testid="timeline-item-1"
                id="1"
              >
                <div
                  class="paperContent"
                >
                  <div>
                    Testing
                  </div>
                  <div
                    class="user"
                  >
                    <div
                      class="avatarIconContainer"
                      data-testid="avatar-icon"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root userButton MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                        data-testid="timeline-avatar"
                        disabled=""
                        id="timeline-avatar"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root userAvatar"
                            data-testid="avatar-icon-avatar"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    nsawas
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
